javascript/apollo/security/apollo-axios-ssrf.yaml

```diff
@@ -0,0 +1,21 @@
+rules:
+- id: apollo-axios-ssrf
+  languages:
+  - js
+  message: User-controllable argument $DATAVAL to $METHOD passed to Axios via internal handler $INNERFUNC
```

Comment:
What vulnerability exists if this happens? What should I do to fix it?

Also, do you have any links to articles or documentation for this particular issue?

This information ends up being immensely helpful when the rule actually triggers. It gives the end user some clear next steps! :D

Reply:
This pattern is indicative of server-side request forgery.  A user may use an affected service to call restricted APIs or leak internal headers to an unauthorized third party.

The fix is to validate user arguments against an allow-list of known URLs or (preferably) to refactor the behavior so a user-controllable URL is not necessary.

https://owasp.org/www-community/attacks/Server_Side_Request_Forgery
https://www.cvedetails.com/cve/CVE-2020-28168/

Reply:
Thanks! I added all this info to the rule :D